freqtrade/freqai/RL/Base4ActionRLEnv.py

```diff
@@ -292,7 +292,8 @@ def _update_profit(self, action):
             pnl = self.get_unrealized_profit()
 
             if self._position in (Positions.Long, Positions.Short):
-                self._total_profit *= (1 + pnl)
+                #self._total_profit *= (1 + pnl)
+                self._total_profit += pnl
```

Comment:
So I agree with this change, but I think we need to check if the user wants to compound or not based on the `self.config['trade_amount'] `.

Reply:
`"stake_amount"` - maybe? 

i'm not sure why this is relevant though - that's relevant to open new positions (is my next position equally sized, or slightly bigger or smaller).

Assuming `pnl` is the absolute profit though - i don't see why this is relevant here.
if it's relative profit - it's wrong anyway (in both forms) , as you can't aggregate percentage values with different stakes.

Reply:
Summing the pnl would assume each trade stake is 1. Which is how the environment is designed. This is why I had it compounding originally - it uses the full "trade wallet" (but only for one coin which would be the `stake_amount`). That said...

Looking closer, it is also just plain wrong to `update_profit` this way at *each training step*. This should only be updated when a training trade is closed:

```python 
      if self._position in (Positions.Long, Positions.Short) and action in (Actions.Long_exit.value, Actions.Short_exit.value):
              self._total_profit *= (1 + pnl)
              #self._total_profit *= (1 + pnl)
              self._total_profit += pnl
  ```

Reply:
But PnL itself is unrealized profit, so that wouldn't be to right if we just update it when the trade is closed. Maybe we need another function which is update it when the exit signal is formed, but until that because we use unrealized profit here that is right.

Reply:
Well we can't be summing unrealized profit.

Assume a trade goes perfectly, every tick you sum the unrealized profit at that point during the trade. In the end you add the full profit from the entire trade. It is not correct.

Reply:
Another bug was located in this function - it only updates profit when the action is a trade entry. Also incorrect.

I have corrected this two bugs, added compounding detection, and refactored the environment inheritance tree in another commit. Please check to ensure you still agree with `_update_profit()`, and make a PR if you feel it is still incorrect. But I am closing this PR since the other commit I made covers this and a lot more.